tracer/src/Datadog.Trace/Ci/CIEnvironmentValues.cs

```diff
@@ -382,8 +410,64 @@ internal void ReloadEnvironmentData()
             // **********
             Branch = GetEnvironmentVariableIfIsNotEmpty("DD_GIT_BRANCH", Branch);
             Tag = GetEnvironmentVariableIfIsNotEmpty("DD_GIT_TAG", Tag);
-            Repository = GetEnvironmentVariableIfIsNotEmpty("DD_GIT_REPOSITORY_URL", Repository);
-            Commit = GetEnvironmentVariableIfIsNotEmpty("DD_GIT_COMMIT_SHA", Commit);
+            Repository = GetEnvironmentVariableIfIsNotEmpty(
+                "DD_GIT_REPOSITORY_URL",
+                Repository,
+                (value, defaultValue) =>
+                {
+                    if (value is not null)
+                    {
+                        value = value.Trim();
+                        if (value.Length == 0)
+                        {
+                            Log.Error("DD_GIT_REPOSITORY_URL is set with an empty value, defaulting to '{default}'", defaultValue);
+                        }
+                        else if (Regex.Match(value, RepositoryUrlPattern).Length != value.Length)
+                        {
+                            Log.Error("DD_GIT_REPOSITORY_URL is set with an invalid value ('{value}'), defaulting to '{default}'", value, defaultValue);
+                        }
+                        else
+                        {
+                            // All ok!
+                            return true;
+                        }
+                    }
+
+                    if (string.IsNullOrEmpty(defaultValue))
+                    {
+                        Log.Error("The Git repository couldn't be automatically extracted.");
+                    }
+
+                    // If not set use the default value
+                    return false;
+                });
+            Commit = GetEnvironmentVariableIfIsNotEmpty(
+                "DD_GIT_COMMIT_SHA",
+                Commit,
+                (value, defaultValue) =>
+                {
+                    if (value is not null)
+                    {
+                        value = value.Trim();
+                        if (value.Length < 40 || !IsHex(value))
+                        {
+                            Log.Error("DD_GIT_COMMIT_SHA must be a full-length git SHA, defaulting to '{default}'", defaultValue);
+                        }
```

Comment:
Same here, would be nice to not have 2 errors when wrong sha and failed to extract, but up to you 🙂

Reply:
done in [5f6129b](https://github.com/DataDog/dd-trace-dotnet/pull/3560/commits/5f6129b79c10083f07e4d9454611bf42eb428b22)